tools/build_defs/repo/git.bzl

```diff
@@ -53,8 +53,13 @@ def _clone_or_update(ctx):
 set -ex
 ( cd {working_dir} &&
     if ! ( cd '{dir_link}' && [[ "$(git rev-parse --git-dir)" == '.git' ]] ) >/dev/null 2>&1; then
-      rm -rf '{directory}' '{dir_link}'
```

Comment:
Unconditionally dropping the removal of `'{dir_link}'` causes the later `ctx.symlink(dest_link, ctx.path("."))` to fail (on all platforms, including windows). Please change the request in such a way that the existings tests, in partiuclar `//src/test/shell/bazel:skylark_git_repository_test` continue to pass.